### ISSUE TITLE
[Google3 Bug]: [a11y] [iOS only] Voiceover doesn't automatically scroll screen inside bottom sheet with keyboard open

### ISSUE THREAD

**Issue by pedrovsantos-lab** [flutter/flutter#174517]
### Help us understand the severity of this issue

- [ ] causing severe production issues e.g. malfunctions or data loss
- [ ] blocking next binary release
- [ ] blocking a client feature launch within a quarter
- [ ] nice-to-have but does not block a launch within the next quarter

### Steps to reproduce

1. Have a bottom sheet that can occupy full screen and is scrollable
2. Have an action in the bottom sheet that opens a keyboard
3. Try to scroll to a button in the end of the page that is below the keyboard

### Expected results

VoiceOver focus and shows the buttons as expected, scrolling the keyboard as needed

### Actual results

VoiceOver is focusing on the buttons but the scroll doesn't hide the keyboard

### Code sample

N/A

### Screenshots or Video

See b/427918745 for details. Works on Android.

### Flutter Doctor output

N/A

**Comment by darshankawar** [comment]
Based on the steps to reproduce, I came up with below code which has textfield in `showModalBottomSheet` along with a textfield and a button which gets hidden behind the keyboard upon tapping the textfield. I then tried using talkback but was unable to verify that it autoscrolls properly to the hidden button.

@pedrovsantos-lab Can you please check the below code and kindly update as required that correctly replicates the reported behavior ?

<details>
<summary> runnable code </summary>

```

import 'package:flutter/material.dart';

/// Flutter code sample for [showModalBottomSheet].

void main() => runApp(const BottomSheetApp());

class BottomSheetApp extends StatelessWidget {
  const BottomSheetApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Bottom Sheet Sample')),
        body: const BottomSheetExample(),
      ),
    );
  }
}

class BottomSheetExample extends StatelessWidget {
  const BottomSheetExample({super.key});

  @override
  Widget build(BuildContext context) {
    return Center(
      child: ElevatedButton(
        child: const Text('showModalBottomSheet'),
        onPressed: () {
          showModalBottomSheet<void>(
          //  isScrollControlled: true,
            context: context,
            builder: (BuildContext context) {
              return Container(
                height: 900,
                color: Colors.amber,
                child: Center(
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    mainAxisSize: MainAxisSize.min,
                    children: <Widget>[
                      const Text('Modal BottomSheet'),
                      ElevatedButton(
                        child: const Text('Close BottomSheet'),
                        onPressed: () => Navigator.pop(context),
                      ),
                      TextField(),
                      ElevatedButton(
                        child: Text('This button is behind the keyboard'),
                        onPressed: () {},
                      )
                    ],
                  ),
                ),
              );
            },
          );
        },
      ),
    );
  }
}

```

</details>
